### DISCUSSION TITLE
How to make a MacOS (Installer) DMG?

### DISCUSSION THREAD

**Discussion by JaneX8** [wailsapp/wails#4192] [Q&A]
### Have you read the Documentation Contribution Guidelines?

- [X] I have read the [Documentation Contribution Guidelines](https://wails.io/community-guide#ways-of-contributing).

### Description

I reluctantly purchased an M1 Apple device simply for building Wails app in the future. I have setup Homebrew, Go, Wails, Xcode and managed to build a universal .app file. Thank you for that great work. Now I only need to figure out how to test the universal app on a non-ARM Mac. And also, how to build an DMG Installer for Mac.

I've found some references to DMG files but none in the documentation and none offer a solution:
- https://github.com/wailsapp/wails/issues/1292#issue-1183456635 (most relevant)
- https://github.com/wailsapp/wails/issues/1178#issuecomment-1341942610 (less relevant)
- https://github.com/wailsapp/wails/issues/485#issue-689272867 (irrelevant)

### Self-service

- [X] I'd be willing to address this documentation request myself.

**Comment by leaanthony** [comment]
There's a node utility that does a pretty good job. There's no native go way currently (not if you want a working .DS_Store file). Mat has a good write up on the process here https://medium.com/@mattholt/packaging-a-go-application-for-macos-f7084b00f6b5

**Comment by JaneX8** [comment]
I just figured out the basics by making this script. But I like the background in Mat's approach, I'll check that out too. Thanks.

After running `wails build -platform darwin/universal -clean` and setting the corrects paths I made it using:

```bash
# Set variables
APP_PATH="/Users/xx/wails-test/wails-test/build/bin/wails-test.app"
DMG_PATH="/Users/xx/wails-test/wails-test/build/bin/wails-test.dmg"
TMP_DIR="/tmp/dmg-build"
VOLUME_NAME="Wails Test"

# Clean up any existing temporary directory
rm -rf "$TMP_DIR"
mkdir -p "$TMP_DIR"

# Copy the .app file to the temporary directory
cp -R "$APP_PATH" "$TMP_DIR/"

# Create the Applications shortcut
ln -s /Applications "$TMP_DIR/Applications"

# Create the DMG file
hdiutil create -volname "$VOLUME_NAME" \
    -srcfolder "$TMP_DIR" \
    -ov \
    -format UDZO \
    "$DMG_PATH"

# Clean up temporary directory
rm -rf "$TMP_DIR"

# Output result
echo "DMG created at: $DMG_PATH"
```

I think it would be great to have a generic "DMG generation" option when using `wails build` on or for macOS to make macOS 'Installers'.

**Comment by JaneX8** [comment]
> There's a node utility that does a pretty good job. There's no native go way currently (not if you want a working .DS_Store file). Mat has a good write up on the process here https://medium.com/@mattholt/packaging-a-go-application-for-macos-f7084b00f6b5

For future reference this was mentioned in the article: https://gist.github.com/mholt/11008646c95d787c30806d3f24b2c844.

**Comment by uqix** [comment]
Just found another wrapper of `hdiutil`: https://github.com/create-dmg/create-dmg
